packages/svelte/src/internal/client/operations.js

```diff
@@ -193,15 +198,15 @@ export function child_frag(node, is_text) {
 		// if an {expression} is empty during SSR, there might be no
 		// text node to hydrate — we must therefore create one
 		if (is_text && first_node?.nodeType !== 3) {
-			const text = document.createTextNode('');
+			const text = empty();
 			current_hydration_fragment.unshift(text);
 			if (first_node) {
 				/** @type {DocumentFragment} */ (first_node.parentNode).insertBefore(text, first_node);
 			}
 			return text;
 		}
 
-		if (first_node !== null) {
+		if (first_node) {
```

Comment:
I believe the explicit comparison is deliberate, since it results in more efficient bytecode

```suggestion
		if (first_node !== null) {
```

Reply:
I did change it because the comparison will be wrongfully false when `first_node` is `undefined`